### PR TITLE
add update-alternatives hooks for deb packages

### PR DESCRIPTION
**Pull request by xpufx**
Tested on ubuntu 20.04/21.10 and Linux Mint 20.3

Please feel free to make changes or apply the idea yourself. No credit needed.

### CODE REVIEW

File: ci/deploy.sh [hunk 1]
```diff
@@ -198,6 +214,10 @@ EOF
         install -Dsm755 -t pkg/debian/usr/bin target/release/strip-ansi-escapes
 
         deps=$(cd pkg && dpkg-shlibdeps -O -e debian/usr/bin/*)
+	      mv pkg/debian/postinst pkg/debian/DEBIAN/postinst
```
Comment (wez) [inline ci/deploy.sh:217]: indent looks a bit weird here; tabs vs. spaces?

File: ci/deploy.sh [hunk 2]
```diff
@@ -198,6 +214,10 @@ EOF
...
+        chmod 0555 pkg/debian/DEBIAN/postinst
```
Comment (wez) [inline ci/deploy.sh:218]: These `0555` should probably be `0755`?  `555` is a bit unusual!  Does the deb stuff require it?

**Review by wez — COMMENTED** [review]
Thanks!

File: ci/deploy.sh (hunk 1 above)
Comment (xpufx) [reply ci/deploy.sh:217]: It's my editor. It looked different everywhere. I tried to fix it on the github editor but apparently it's still borked. Sorry.

File: ci/deploy.sh (hunk 2 above)
Comment (xpufx) [reply ci/deploy.sh:218]: Docs says something like 0555 to 0755. I don't think it matters. But we can make it 755 too.

File: ci/deploy.sh (hunk 1 above)
Comment (wez) [inline ci/deploy.sh:217]:
```suggestion
        mv pkg/debian/postinst pkg/debian/DEBIAN/postinst
```

File: ci/deploy.sh (hunk 2 above)
Comment (wez) [inline ci/deploy.sh:218]:
```suggestion
        chmod 0755 pkg/debian/DEBIAN/postinst
```

File: ci/deploy.sh [hunk 3]
```diff
@@ -198,6 +214,10 @@ EOF
...
+        mv pkg/debian/prerm pkg/debian/DEBIAN/prerm
+        chmod 0555 pkg/debian/DEBIAN/prerm
```
Comment (wez) [inline ci/deploy.sh:220]:
```suggestion
        chmod 0755 pkg/debian/DEBIAN/prerm
```

File: ci/deploy.sh [hunk 4]
```diff
@@ -190,6 +190,22 @@ Description: Wez's Terminal Emulator.
...
+
+        cat > pkg/debian/postinst <<EOF
+#!/bin/sh
+set -e
+if [ "\$1" = "configure" ] ; then
+        update-alternatives --install /usr/bin/x-terminal-emulator x-terminal-emulator /usr/bin/wezterm 20
+fi
+EOF
+
+	      cat > pkg/debian/prerm <<EOF
```
Comment (wez) [inline ci/deploy.sh:203]:
```suggestion
        cat > pkg/debian/prerm <<EOF
```

**Comment by wez** [comment]
Thanks!

**Pull request merged**